Add tutorials guidelines to the main tutorial branch (#2458)
pyansys/pydpf-core: doc/source/getting_started/contribute/documentarian.rst

Contributing as a documentarian
###############################

.. note::

    Overall guidance on contributing to the documentation of a PyAnsys repository appears in
    `Documenting`_ in the *PyAnsys Developer's Guide*.

    You must also follow the `Documentation style`_ guide to
    ensure that all the documentation looks the same across the project.

To contribute on the documentation you must start by setting up the PyDPF-Core repository
by following the steps in :ref:`contributing_as_a_developer` section.

In this page you can check how to :

.. grid:: 1 2 3 3
    :padding: 2 2 2 2

    .. grid-item-card:: :fa:`th` Structure the documentation
        :link: structure-documentation
        :link-type: ref

        How the documentation is structured and where to locate files.

    .. grid-item-card:: :fa:`pencil` Write documentation
        :link: write-product-use-documentation
        :link-type: ref

>         Explains and showcase the use of PyDPF-Core.

Reviewer: ```suggestion
        Explains and showcases the use of PyDPF-Core.
```

The path to the file is relative to the directory where the table of contents
is declared.

.. _write-product-use-documentation:

Write documentation
===================

Our documentation tries to follow a structure principle that respects four different functions of the documentation.
Each of them fulfills a different need for people working with our tool at different times, in different circumstances.

Here is an overview of how our documentation is organized to help you know where you should include your contributions.
Each section has their own guidelines that must be followed when creating new content.
To check these specific guidelines click on the correspondent card below.

.. grid:: 1 1 2 2
    :gutter: 2
    :padding: 2
    :margin: 2

    .. grid-item-card:: **TUTORIALS**
       :link: ref_guidelines_tutorials
       :link-type: ref
       :class-title: sd-text-center sd-bg-light
       :class-header: sd-text-center

       Learning oriented
       ^^^^^^^^^^^^^^^^^

       **Function:**  Teach how to get started and use PYDPF-core step by step

       They are designed to teach how to perform a task and understand the underlying concepts,

>        providing detailed explanations at each stage. The task is built around the application of specific features.

Reviewer: ```suggestion
       providing detailed explanations at each stage. The tutorials are feature-oriented.
```

    .. grid-item-card:: **EXAMPLES**
       :link: ref_guidelines_examples
       :link-type: ref
       :class-title: sd-text-center sd-bg-light
       :class-header: sd-text-center

       Use-cases oriented
       ^^^^^^^^^^^^^^^^^^

       **Function:**  Show how to solve specifics key problems

>        They showcase specific key problems and use-cases. They are more advanced than

Reviewer: ```suggestion
       Showcase specific key problems and use-cases with a complete PyDPF script. They are more advanced than
```

       tutorials as they present end-to-end engineering workflows and assume basic knowledge of PyDPF-Core.

    .. grid-item-card:: **CONCEPTS**
       :class-title: sd-text-center sd-bg-light
       :class-header: sd-text-center

       Understanding oriented
       ^^^^^^^^^^^^^^^^^^^^^^

       **Function:**  Provide useful theoretical explanations for PyDPF-Core

       They discuss and explain key DPF principles and concepts, enabling the reader to understand the spirit of the underlying tool.

    .. grid-item-card:: **API REFERENCE**
       :class-title: sd-text-center sd-bg-light
       :class-header: sd-text-center

       Informing oriented
       ^^^^^^^^^^^^^^^^^^

       **Function:** Describe PyDPF-Core APIs

>        They contain technical reference on how PyDPF-Core works and how to use it but assume basic
>        understanding of key DPF concepts. It is generated automatically along the documentation and
>        is based on the source code.

Reviewer: ```suggestion
       Provides technical reference on how PyDPF-Core works and how to use it but assume basic
       understanding of key DPF concepts. It is generated automatically along the documentation and
       is based on the source code.
```
